concent_api/core/tests/utils.py

```diff
@@ -197,13 +202,57 @@ def _test_database_objects(
         task_id                     = None,
         receive_delivered_status    = None,
     ):
-        self.assertEqual(StoredMessage.objects.last().type,           last_object_type.TYPE)
-        self.assertEqual(StoredMessage.objects.last().task_id,        task_id)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().type,      last_object_type.TYPE)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().task_id,   task_id)
 
         if receive_delivered_status is not None:
             self.assertEqual(ReceiveStatus.objects.last().delivered,        receive_delivered_status)
             self.assertEqual(ReceiveStatus.objects.last().message.task_id,  task_id)
 
+    def _test_subtask_state(
+        self,
+        task_id,
+        subtask_id,
+        subtask_state,
+        provider_key,
+        requestor_key,
+        next_deadline                   = None,
+        has_task_to_compute             = None,
+        has_report_computed_task        = None,
+        has_ack_report_computed_task    = None,
+        has_reject_report_computed_task = None,
+        has_subtask_results_accepted    = None,
+        has_subtask_results_rejected    = None,
+    ):
+        self.assertTrue(StoredMessage.objects.filter(subtask_id = subtask_id).exists())
+        subtask = Subtask.objects.get(subtask_id = subtask_id)
+        self.assertEqual(subtask.task_id,              task_id)
+        self.assertEqual(subtask.subtask_id,           subtask_id)
+        self.assertEqual(subtask.state,                subtask_state)
+        self.assertEqual(subtask.provider.public_key,  provider_key)
+        self.assertEqual(subtask.requestor.public_key, requestor_key)
+
+        self.assertTrue(Client.objects.filter(public_key = provider_key).exists())
+        self.assertTrue(Client.objects.filter(public_key = requestor_key).exists())
```

Comment:
These two should be normal `assert`s. This looks like a sanity check, not something that can really go wrong in the test. The fact that there is a `Client` instance in the database with a `public_key` used in a `Subtask` is guaranteed because `provider` and `requestor` are foreign keys. Basically, this could only happen if there's something seriously wrong the the model.

Reply:
:+1: